### ISSUE TITLE
Add commonAnnotations configuration option for kustomize 

### ISSUE THREAD

**Issue by LeonardAukea** [argoproj/argo-cd#4603]
# Summary

I noticed in the [docs](https://argoproj.github.io/argo-cd/user-guide/kustomize/
) that you currently only support: 

* `namePrefix` is a prefix appended to resources for Kustomize apps
* `nameSuffix` is a suffix appended to resources for Kustomize apps
* `images` is a list of Kustomize image overrides
* `commonLabels` is a string map of an additional labels 

https://argoproj.github.io/argo-cd/user-guide/kustomize/

I think there is a need to also support `commonAnnotations` as well:  https://kubectl.docs.kubernetes.io/pages/reference/kustomize.html#commonannotations

# Motivation

As an example, we need to add the following annotations to add vault auth as a sidecar. 

```yaml 
commonAnnotations:
  injector.tumblr.com/request: vault-auth
```

I was actually surprised to find out that there was no support for this and learnt it the hard way :/ 

# Proposal

Very similarly as to how `commonLabels` is currently implemented

**Comment by LeonardAukea** [comment]
@jannfis Could you provide any pointers as to where the related code might reside. eg. where is the implementation for `commonLabels` and if this is potentially manageable for a non go developer I would be willing to give it a shot

**Comment by jannfis** [comment]
Implemented by #4613 

Sorry @LeonardAukea, seems @igaskin was quicker sending in a PR than me answering your question.

**Comment by LeonardAukea** [comment]
Next time @jannfis :)
